go.mod: do not lie about googleproto version

tbg opened the pull request:
It seems to "just work" after also upgrading `github.com/golang/protobuf`. The third commit optimistically upgrades `google.golang.org/protobuf` to the latest version v1.35.1.

Fixes #134941.

Epic: none

tbg committed:
go.mod: do not lie about googleproto version

Doesn't build:

> external/com_github_golang_protobuf/protoc-gen-go/descriptor/descriptor.pb.go:106:61:
> undefined: descriptorpb.Default_FileOptions_PhpGenericServices

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/134943)
<!-- Reviewable:end -->

tbg committed:
go.mod: go get github.com/golang/protobuf

https://github.com/flipt-io/flipt/pull/2841#issuecomment-2004075375

Release Note: none
Epic: none

tbg committed:
go.mod: go get google.golang.org/protobuf@v1.35.1

Epic: none
Release note: None

tbg commented:
TFTRs!

bors r+
